src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,297 @@
 #include <ripple/app/reporting/P2pProxy.h>
 #include <ripple/json/json_value.h>
 #include <ripple/net/RPCErr.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/SField.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFormats.h>
+#include <ripple/protocol/digest.h>
 #include <ripple/protocol/jss.h>
 #include <ripple/rpc/Context.h>
 #include <ripple/rpc/Role.h>
-#include <ripple/rpc/impl/TransactionSign.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <unordered_map>
 
 namespace ripple {
 
+namespace detail {
+
+class ServerDefinitions
+{
+private:
+    std::string
+    translate(std::string const& inp);
+
+    Json::Value
+    generate();
+
+    uint256 defsHash_;
+    Json::Value defs_;
+
+public:
+    ServerDefinitions() : defs_(generate()){};
+
+    bool
+    hashMatches(uint256 hash) const
+    {
+        return defsHash_ == hash;
+    }
+
+    Json::Value const&
+    get() const
+    {
+        return defs_;
+    }
+};
+
+std::string
+ServerDefinitions::translate(std::string const& inp)
+{
+    auto replace = [&](char const* oldStr, char const* newStr) -> std::string {
+        std::string out = inp;
+        boost::replace_all(out, oldStr, newStr);
+        return out;
+    };
+
+    auto contains = [&](char const* s) -> bool {
+        return inp.find(s) != std::string::npos;
+    };
+
+    if (contains("UINT"))
+    {
+        if (contains("256") || contains("160") || contains("128"))
+            return replace("UINT", "Hash");
+        else
+            return replace("UINT", "UInt");
+    }
+
+    std::unordered_map<std::string, std::string> replacements{
+        {"OBJECT", "STObject"},
+        {"ARRAY", "STArray"},
+        {"ACCOUNT", "AccountID"},
+        {"LEDGERENTRY", "LedgerEntry"},
+        {"NOTPRESENT", "NotPresent"},
+        {"PATHSET", "PathSet"},
+        {"VL", "Blob"},
+        {"XCHAIN_BRIDGE", "XChainBridge"},
+    };
+
+    if (auto const& it = replacements.find(inp); it != replacements.end())
+    {
+        return it->second;
+    }
+
+    std::string out;
+    size_t pos = 0;
+    std::string inpToProcess = inp;
+    for (;;)
```

Comment:
```suggestion
    // this loop converts "snake_case" types into "camelCase"
    for (;;)
```

Reply:
Done in https://github.com/XRPLF/rippled/pull/4703/commits/205c0f5bc37b99d22becd640c4f587114f961edc